utils/doclint/generateDotnetApi.js

```diff
@@ -0,0 +1,695 @@
+          // for now, we'll merge the two enums, if they have the same name, and we'll go from there
+          potentialEnum.concat(enumValues.filter(x => !potentialEnum.includes(x))); // merge & de-dupe
+          // TODO: think about doing global type annotation, where we can add comments, such as this?
+          enumTypes.set(enumName, potentialEnum);
+        }
+      } else {
+        enumTypes.set(enumName, enumValues);
+      }
+      if (isNullableEnum)
+        return `${enumName}?`;
+      return enumName;
+    }
+
+    if (type.expression === '[string]|[Buffer]')
+      return `byte[]`; // TODO: make sure we implement extension methods for this!
+    else if (type.expression === '[string]|[float]'
+      || type.expression === '[string]|[float]|[boolean]'
+      || (type.union && type.union[0].name === 'string' && type.union[1].name === 'float' && type.union[2].name === 'boolean'))
+      return 'string'; // because users can send either 100 or 100px for most of these arguments
+    else if (type.union.length == 2 && type.union[1].name === 'Array' && type.union[1].templates[0].name === type.union[0].name)
+      return `IEnumerable<${type.union[0].name}>`; // an example of this is [string]|[Array]<[string]>
+    else if (type.union[0].name === 'path')
+      // we don't support path, but we know it's usually an object on the other end, and we expect
+      // the dotnet folks to use [NameOfTheObject].LoadFromPath(); method which we can provide separately
+      return translateType(type.union[1], parent, generateNameCallback);
+    else if (type.expression === '[float]|"raf"')
+      return `Polling`; // hardcoded because there's no other way to denote this
+
+    return null;
+  }
+
+  if (type.name === 'Array') {
+    if (type.templates.length != 1)
+      throw `Array (${type.name} from ${parent.name}) has more than 1 dimension. Panic.`;
+
+    let innerType = translateType(type.templates[0], parent, generateNameCallback);
+    return `IEnumerable<${innerType}>`;
+  }
+
+  if (type.name === 'Object') {
+    // take care of some common cases
+    // TODO: this can be genericized
+    if (type.templates && type.templates.length == 2) {
+      // get the inner types of both templates, and if they're strings, it's a keyvaluepair string, string, 
+      let keyType = translateType(type.templates[0], parent, generateNameCallback);
+      let valueType = translateType(type.templates[1], parent, generateNameCallback);
+      return `IEnumerable<KeyValuePair<${keyType}, ${valueType}>>`;
+    }
+
+    if ((type.name === 'Object')
+      && !type.properties
+      && !type.union) {
+      return 'object';
+    }
+    // this is an additional type that we need to generate
+    let objectName = generateNameCallback(type);
+    if (objectName === 'Object') {
+      throw 'Object unexpected';
+    } else if (type.name === 'Object') {
+      registerAdditionalType(objectName, type);
+    }
+
+    if (['object'].includes(objectName))
```

Comment:
Are both of these cases possible (objectName === 'Object' ` above and `objectName === 'object' ` here) ?

Reply:
This looks like a remnant of a copy-paste/refactoring effort :D